### PR TITLE
Authentication 

### PR DESCRIPTION
**Pull request by JonathanWitman**
Login, user authentication, and logout.

### CODE REVIEW

File: portal/__init__.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
-from flask import Flask, render_template
+from flask import Flask, render_template, request, redirect, url_for, session, g
```
Comment (zachfedor) [inline portal/__init__.py]: extra imports, clean these up

File: portal/auth.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+@bp.route('/', methods=['GET', 'POST'])
+def index():
+    if request.method == 'POST':
+
+        email  = request.form['email']
+        password  = request.form['password']
+
+        if email is None or password is None:
+            return render_template('index.html')
```
Comment (zachfedor) [inline portal/auth.py]: bad UX since there's no feedback for user

File: portal/auth.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            conn = db.get_db()
+            cur = conn.cursor()
+            cur.execute("SELECT * FROM users WHERE email = %s;", (email,))
+            user = cur.fetchone()
+
+            if user is None:
+                return redirect(url_for('index'))
+            # utilize hashes and compare hashed password to password in DB
+            elif not check_password_hash(user[2], password):
+                return redirect(url_for('index'))
```
Comment (zachfedor) [inline portal/auth.py]: remove redirect

File: portal/static/styles.css [hunk 4]
```diff
@@ -10,12 +10,30 @@
...
+.header h1 {
+  font-family: serif;
   font-weight: 400;
-  color: red;
+  background-color: rgb(128, 40, 65);
+  color: #fafafa;
+  width: 100%;
+  margin: 0;
+  padding: 1em 0 1em .5em;
+  font-family: 'Libre Baskerville', serif;
```
Comment (zachfedor) [inline portal/static/styles.css:30]: stick to non-existent (yet) style guide

File: tests/conftest.py [hunk 5]
```diff
@@ -37,3 +37,21 @@ def client(app):
 def runner(app):
     return app.test_cli_runner()
 
+
+class AuthActions(object):
+    def __init__(self, client):
+        self._client = client
+
+    def login(self, email='test', password='test'):
```
Comment (zachfedor) [inline tests/conftest.py]: this won't work by default, might want to add better defaults

**Review by zachfedor — CHANGES_REQUESTED** [review]

**Review by Lnauman1700 — APPROVED** [review]
👍

**Review by zachfedor — APPROVED** [review]

**Pull request merged**